### DISCUSSION TITLE
`com.google.fonts/check/legacy_accents` rationale question

### DISCUSSION THREAD

**Discussion by bobh0303** [fonttools/fontbakery#4555] [Q&A] [answered]
The rationale for the `com.google.fonts/check/legacy_accents` check says:

> Legacy accents should not be used in accented glyphs. 

that is, should not be used as a component in composite glyphs that represent accented characters.

>The use of legacy
> accents in accented glyphs breaks the mark to mark combining feature that
> allows a font to stack diacritics over one glyph. 

Can some explain to me how use of legacy accent glyphs _as components_ of another glyph "breaks the mark to mark combining feature"? That composite glyph has its own anchors, it's own GDEF class value, and its own advance width -- all independent of the attributes of the component glyphs themselves -- so what is the problem?

**Comment by simoncozens** [comment, marked as answer]
This check combines multiple ideas. (I have to admit I don't really like it.) One of the ideas is about how fonts are constructed by the designer - we think it is good practice that they use combining accents because they're more likely to get the anchors right, whereas legacy accents should not have anchors. And OK, I can kind of see that. But the other idea is how these things end up in the binary font, which I don't think is relevant at all. See further discussion in #3959.

**Reply by moyogo** [reply]
> Can some explain to me how use of legacy accent glyphs as components of another glyph "breaks the mark to mark combining feature"?

It doesn’t. This check is broken.

> That composite glyph has its own anchors, it's own GDEF class value, and its own advance width -- all independent of the attributes of the component glyphs themselves -- so what is the problem?

This check is the problem. What you’re doing is good practice.

> we think it is good practice that they use combining accents because they're more likely to get the anchors right, whereas legacy accents should not have anchors. 

Then legacy accents or more accurately spacing marks should not have mark anchors. That’s what the check should ensure is not happening instead.

**Reply by felipesanches** [reply]
The check was temporarily disabled now via PR https://github.com/fonttools/fontbakery/pull/4567

**Comment by bobh0303** [comment]
And thank you that this works:
```
    "overrides": {
        "com.google.fonts/check/legacy_accents": [
            {
                "code": "legacy-accents-component",
                "status": "PASS",
                "reason": "Generates unnecessary WARNs in our workflow.",
            },
        ],
    },
```
